From b6ebeeac71aba19031ac94ed8f38ac1f180f5170 Mon Sep 17 00:00:00 2001
From: Patrick Mooney <pmooney@pfmooney.com>
Date: Fri, 18 Nov 2016 21:42:08 +0000
Subject: [PATCH] OS-5808 lxbrand leaks cpuset_t from lwp data Reviewed by:
 Jerry Jelinek <jerry.jelinek@joyent.com> Reviewed by: Ryan Zezeski
 <rpz@joyent.com> Approved by: Ryan Zezeski <rpz@joyent.com>

---
 usr/src/uts/common/brand/lx/os/lx_misc.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/usr/src/uts/common/brand/lx/os/lx_misc.c b/usr/src/uts/common/brand/lx/os/lx_misc.c
index e2043a9ab6..b8c9b52329 100644
--- a/usr/src/uts/common/brand/lx/os/lx_misc.c
+++ b/usr/src/uts/common/brand/lx/os/lx_misc.c
@@ -328,6 +328,13 @@ lx_freelwp(klwp_t *lwp)
 		lx_pid_rele(lwptoproc(lwp)->p_pid, lwptot(lwp)->t_tid);
 	}
 
+	/*
+	 * Discard the affinity mask.
+	 */
+	VERIFY(lwpd->br_affinitymask != NULL);
+	cpuset_free(lwpd->br_affinitymask);
+	lwpd->br_affinitymask = NULL;
+
 	/*
 	 * Ensure that lx_ptrace_exit() has been called to detach
 	 * ptrace(2) tracers and tracees.
-- 
2.21.0

